"""
Generated Protocol Buffer definitions for Omega-RAG Episode schema.
"""

from .episode_pb2 import (
    Episode,
    Problem,
    Plan,
    SubTask,
    Trajectory,
    ExecutionStep,
    StepStatus,
    Outcome,
    EpisodeMetadata,
    StoreEpisodeRequest,
    StoreEpisodeResponse,
    GetEpisodeRequest,
    GetEpisodeResponse,
    SearchEpisodesRequest,
    SearchEpisodesResponse,
    EpisodeSearchResult,
    EpisodeFilter,
    QueryEmbedding,
    BatchStoreEpisodesRequest,
    BatchStoreEpisodesResponse,
    UpdateEpisodeMetadataRequest,
    UpdateEpisodeMetadataResponse,
    DeleteEpisodeRequest,
    DeleteEpisodeResponse,
    EpisodeStatistics,
    GetStatisticsRequest,
    GetStatisticsResponse,
)

from .filter_helpers import create_episode_filter, has_filter_value

__all__ = [
    "Episode",
    "Problem",
    "Plan",
    "SubTask",
    "Trajectory",
    "ExecutionStep",
    "StepStatus",
    "Outcome",
    "EpisodeMetadata",
    "StoreEpisodeRequest",
    "StoreEpisodeResponse",
    "GetEpisodeRequest",
    "GetEpisodeResponse",
    "SearchEpisodesRequest",
    "SearchEpisodesResponse",
    "EpisodeSearchResult",
    "EpisodeFilter",
    "QueryEmbedding",
    "BatchStoreEpisodesRequest",
    "BatchStoreEpisodesResponse",
    "UpdateEpisodeMetadataRequest",
    "UpdateEpisodeMetadataResponse",
    "DeleteEpisodeRequest",
    "DeleteEpisodeResponse",
    "EpisodeStatistics",
    "GetStatisticsRequest",
    "GetStatisticsResponse",
    "create_episode_filter",
    "has_filter_value",
]
